frontend/packages/network-attachment-definition-plugin/src/hooks/network-specific-elements-hook.tsx

```diff
@@ -0,0 +1,174 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import * as classNames from 'classnames';
+import { FormControl, FormGroup, HelpBlock } from 'patternfly-react';
+import { Dropdown } from '@console/internal/components/utils';
+import { ELEMENT_TYPES, NetworkTypeParams, networkTypeParams } from '../constants';
+
+export const useNetworkSpecificElements = (
```

Comment:
hook which returns JSX ? thats not correct, maybe you want to use `React.useCallback` directly in `network-attachment-definitions-form.tsx `

Reply:
`React.useCallback` worked like a charm. =)